pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -887,6 +887,12 @@ protected void doTakeSnapshot() {
         ServerGauge.UPSERT_VALID_DOC_ID_SNAPSHOT_COUNT, numImmutableSegments);
     _serverMetrics.setValueOfPartitionGauge(_tableNameWithType, _partitionId,
         ServerGauge.UPSERT_PRIMARY_KEYS_IN_SNAPSHOT_COUNT, numPrimaryKeysInSnapshot);
+    int numMissedSegments = numTrackedSegments - numImmutableSegments - numConsumingSegments;
+    if (numMissedSegments > 0) {
+      _serverMetrics.setValueOfPartitionGauge(_tableNameWithType, _partitionId,
```

Comment:
Use a counter instead. Once a gauge is set, it will stay at that value unless removed.

Reply:
Ah, good call!

Reply:
Updated to Meter!